### PR TITLE
25 events

### PR DESCRIPTION
**Pull request by geckoslair**

### CODE REVIEW

File: lib/paratii.eth.events.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        subscription = this.subscribe('logs', options, listener)
+        break
+      default:
+        if (options === undefined) {
+          options = {
+            fromBlock: null,
+            topics: this._topics[eventType]
+          }
+        }
+        subscription = this.subscribe('logs', options, listener)
```
Comment (jellegerbrandy) [inline lib/paratii.eth.events.js]: We want to subscribe to the logs of a specific contract. I.e. to the `Transfer` event of the contract that is on `paratii.eth.getContractAddress('ParatiiToken')`. I think that works by settings `options.address` to this address : http://web3js.readthedocs.io/en/1.0/web3-eth-subscribe.html#id16

File: test/paratii.eth.events.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  it('subscription to Tranfer PTI events should work as expected', async function () {
+    let beneficiary = '0xDbC8232Bd8DEfCbc034a0303dd3f0Cf41d1a55Cf'
+    let ptiTransfer = paratii.eth.events.addListener('transfer')
+
+    let amount = paratii.eth.web3.utils.toWei('4', 'ether')
+    await paratii.eth.transfer(beneficiary, amount, 'PTI')
+
+    ptiTransfer.on('data', function (log) {
+      const received = paratii.eth.web3.utils.hexToNumberString(log.data)
+      assert.equal(received, amount)
```
Comment (jellegerbrandy) [inline test/paratii.eth.events.js]: I don't think that the test ever arrives at this assertion. You are not really testing anything here.

File: test/paratii.eth.events.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      address: address,
+      privateKey: privateKey
+    })
+    await paratii.eth.deployContracts()
+    paratii.eth.web3.setProvider('ws://localhost:8546/rpc/')
+  })
+
+  it('subscription to Tranfer PTI events should work as expected', async function () {
+    let beneficiary = '0xDbC8232Bd8DEfCbc034a0303dd3f0Cf41d1a55Cf'
+    let ptiTransfer = paratii.eth.events.addListener('transfer')
```
Comment (jellegerbrandy) [inline test/paratii.eth.events.js]: The event is called `Transfer`, not `transfer`.

File: .eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
+{
```
[the remark this answers is not in the document]
Comment (geckoslair) [reply .eslintrc.json]: removed

File: lib/paratii.eth.events.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
+export class ParatiiEthEvents {
+  constructor (context) {
+    this._subscriptions = {}
+    this.subscribe = context.web3.eth.subscribe
+    this._topics = {
+      transfer: ['0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef']
```
[the remark this answers is not in the document]
Comment (geckoslair) [reply lib/paratii.eth.events.js]: this the signature of the event, in order to specialize logs

File: lib/paratii.eth.events.js (hunk 1 above)
Comment (geckoslair) [reply lib/paratii.eth.events.js]: it is not enough, we want to subscribe to particular events inside a contract.

File: test/paratii.eth.events.js (hunk 3 above)
Comment (geckoslair) [reply test/paratii.eth.events.js]: done!

File: test/paratii.eth.events.js (hunk 2 above)
Comment (geckoslair) [reply test/paratii.eth.events.js]: fixed, i've add a done callback, now it's testing something =) sorry!

File: lib/paratii.eth.events.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
+export class ParatiiEthEvents {
+  constructor (context) {
+    this._subscriptions = {}
```
[the remark this answers is not in the document]
Comment (geckoslair) [reply lib/paratii.eth.events.js]: fixed

File: lib/paratii.eth.events.js (hunk 1 above)
Comment (jellegerbrandy) [reply lib/paratii.eth.events.js]: We want to do _both_. So in **addition** to what you are doing here (subscribing to a particular topic), we **also** want to specify that this Event must have logged by a certain specific contact instance. (I.e. we do not want all Transfer events, just those that come from our ParatiiToken)

**Pull request merged**